### DISCUSSION TITLE
[✨] Create a lint rule for `sync$()`

### DISCUSSION THREAD

**Discussion by mhevery** [QwikDev/qwik-evolution#266] [Proposals For Qwik]
### Is your feature request related to a problem?

Qwik now has a new feature, `sync$()`, however, it comes with limitations. It would be great if there was a lint rule which could flag when the developer is using it improperly

### Describe the solution you'd like

A lint rule `qwik/sync$` which would flag whenever the function inside the `sync$()` would close over any variables (other than browser API) and flag it.

### Describe alternatives you've considered

We will create a Rust check as well, but that one will not be instant, and can't provide that much detail.

### Additional context

```typescript
import {foo} from '...

<button onClick$(sync$(() => {
    foo(); // <= lint error. Can't close over 'foo'
  }))>
    click
  </button>
```

**Comment by Lemon2311** [comment]
Hello, I`d like to fix this issue.

**Comment by gioboa** [comment]
Thanks @Lemon2311 🎉 if you need an help don't hesitate to contact me 👍

**Comment by ScriptShah** [comment]
hello, i am new to Open Source i'd like to work on this please assign me.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
